Rakefile

```diff
@@ -31,9 +31,14 @@ namespace :puppet_versions do
     response = http.request(request)
     raise unless response.is_a?(Net::HTTPSuccess)
 
-    File.open(PUPPET_VERSIONS_PATH, 'wb:UTF-8') do |fd|
-      fd.write(JSON.pretty_generate(JSON.parse(response.body)))
+    data = JSON.parse(response.body).filter_map do |_, versions|
+      if !versions['puppet'].nil? && !versions['facter'].nil?
+        [versions['puppet'], versions['facter']]
+      end
     end
+    data.sort_by! { |puppet, _facter| Gem::Version.new(puppet) }.reverse!
```

Comment:
JSON doesn't have a trailing comma so this is sorted from newest to oldest, resulting in a there's a smaller diff on updates.